network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -1121,4 +1305,256 @@ public Optional<Resource<IdentifiableAttributes>> getIdentifiable(UUID networkUu
+                    temporaryLimit.setLimitType(TemporaryLimitType.getByValue(resultSet.getString(6)));
+                    temporaryLimit.setName(resultSet.getString(7));
+                    temporaryLimit.setValue(resultSet.getDouble(8));
+                    temporaryLimit.setAcceptableDuration(resultSet.getInt(9));
+                    temporaryLimit.setFictitious(resultSet.getBoolean(10));
+
+                    temporaryLimits.add(temporaryLimit);
+                }
+                return temporaryLimits;
+            }
+        } catch (SQLException e) {
+            throw new UncheckedSqlException(e);
+        }
+    }
+
+    private List<TemporaryCurrentLimitAttributes> internalPrepareTemporaryLimits(UUID networkUuid,
+                                                                                 int variantNum,
+                                                                                 int side,
+                                                                                 String equipmentId,
+                                                                                 String equipmentType,
+                                                                                 TemporaryLimitType limitType,
+                                                                                 LimitsAttributes limitsAttributes) {
+        ArrayList<TemporaryCurrentLimitAttributes> result = new ArrayList<>();
+        if (limitsAttributes != null && limitsAttributes.getTemporaryLimits() != null) {
+            for (var originalTemporaryLimit : limitsAttributes.getTemporaryLimits().entrySet()) {
+                result.add(TemporaryCurrentLimitAttributes.builder()
+                        .equipmentId(equipmentId)
+                        .equipmentType(equipmentType)
+                        .networkUuid(networkUuid.toString())
+                        .variantNum(variantNum)
+                        .side(side)
+                        .limitType(limitType)
+                        .name(originalTemporaryLimit.getValue().getName())
+                        .value(originalTemporaryLimit.getValue().getValue())
+                        .acceptableDuration(originalTemporaryLimit.getValue().getAcceptableDuration())
+                        .fictitious(originalTemporaryLimit.getValue().isFictitious())
+                        .build());
+            }
+        }
+        return result;
+    }
+
+    private List<TemporaryCurrentLimitAttributes> getTemporaryLimitsFromLines(UUID networkUuid, List<Resource<LineAttributes>> resources) {
+        ArrayList<TemporaryCurrentLimitAttributes> result = new ArrayList<>();
+        if (!resources.isEmpty()) {
+            for (Resource<LineAttributes> resource : resources) {
+                LineAttributes equipment = resource.getAttributes();
+
+                // currentLimits1
+                result.addAll(internalPrepareTemporaryLimits(networkUuid, resource.getVariantNum(), 1,
```

Comment:
Maybe we could use `LimitSelector.getLimits` here ?

Reply:
Yes, this part was refactored.